src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -48,7 +48,7 @@ void symbol_factoryt::gen_nondet_init(
     return;
   }
 
-  if(type.id()==ID_pointer)
+  if(type.id() == ID_pointer && to_pointer_type(type).subtype().id() != ID_code)
```

Comment:
:question: What happens with pointers to code then? Just random number? 
Read the PR description - that is worth a comment here explaining why we don't need to initialize it here

Reply:
Now the symbol is explicitly assigned a nondet (of the function-ptr type).